driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -696,14 +696,15 @@ public void onResult(final D response, final Throwable t) {
         }
     }
 
-    private static boolean shouldAttemptToRetry(@Nullable final BsonDocument command, final Throwable exception,
-                                                final SessionContext sessionContext) {
-        return shouldAttemptToRetry(command != null && command.containsKey("txnNumber"), exception, sessionContext);
+    private static boolean shouldAttemptToRetry(@Nullable final BsonDocument command, final Throwable exception) {
+        return shouldAttemptToRetry(command != null
+                        && (command.containsKey("txnNumber")
+                        || command.getFirstKey().equals("commitTransaction") || command.getFirstKey().equals("abortTransaction")),
```

Comment:
This is a bit lame to check for command names like this, but not sure of a better way to share this code.  It didn't seem much worse than checking for txnNumber.